x-pack/plugins/dashboard_enhanced/public/services/drilldowns/dashboard_to_dashboard_drilldown/drilldown.tsx

```diff
@@ -46,7 +59,105 @@ export class DashboardToDashboardDrilldown
+    config: Config,
+    context: ActionContext<VisualizeEmbeddableContract>
+  ) => {
+    const { navigateToApp } = await this.params.getApplicationService();
+    const dashboardPath = await this.getDestinationUrl(config, context);
+    // note: extracting hash and using 'kibana' as appId will be redundant,
+    // when dashboard move to np urls. (urlGenerator generates np url, which is not supported yet)
+    const dashboardHash = dashboardPath.split('#')[1];
+    await navigateToApp('kibana', {
+      path: `#${dashboardHash}`,
+    });
+  };
+
+  private async getDestinationUrl(
+    config: Config,
+    context: ActionContext<VisualizeEmbeddableContract>
+  ): Promise<string> {
+    const getUrlGenerator = await this.params.getGetUrlGenerator();
+
+    const {
+      createFiltersFromRangeSelectAction,
+      createFiltersFromValueClickAction,
+    } = await this.params.getDataPluginActions();
+    const {
+      timeRange: currentTimeRange,
+      query,
+      filters: currentFilters,
+    } = context.embeddable!.getInput();
+
+    // if useCurrentDashboardFilters enabled, then preserve all the filters (pinned and unpinned)
+    // otherwise preserve only pinned
+    const existingFilters =
+      (config.useCurrentFilters
+        ? currentFilters
+        : currentFilters?.filter(f => esFilters.isFilterPinned(f))) ?? [];
+
+    // if useCurrentDashboardDataRange is enabled, then preserve current time range
+    // if undefined is passed, then destination dashboard will figure out time range itself
+    // for brush event this time range would be overwritten
+    let timeRange = config.useCurrentDateRange ? currentTimeRange : undefined;
+    let filtersFromEvent = await (async () => {
+      try {
+        if (isRangeSelectTriggerContext(context))
+          return await createFiltersFromRangeSelectAction(context.data);
+        if (isValueClickTriggerContext(context))
+          return await createFiltersFromValueClickAction(context.data);
+
+        // eslint-disable-next-line no-console
+        console.warn(
+          `
+          DashboardToDashboard drilldown: can't extract filters from action.
+          Is it not supported action?`,
+          context
+        );
+
+        return [];
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.warn(
+          `
+          DashboardToDashboard drilldown: error extracting filters from action.
+          Continuing without applying filters from event`,
+          e
+        );
+        return [];
+      }
+    })();
+
+    if (context.timeFieldName) {
+      const { timeRangeFilter, restOfFilters } = esFilters.extractTimeFilter(
+        context.timeFieldName,
+        filtersFromEvent
+      );
+      filtersFromEvent = restOfFilters;
+      if (timeRangeFilter) {
+        timeRange = esFilters.convertRangeFilterToTimeRangeString(timeRangeFilter);
+      }
+    }
+
+    return getUrlGenerator(DASHBOARD_APP_URL_GENERATOR).createUrl({
```

Comment:
nit: If we move forward with the recommendation here: https://github.com/elastic/kibana/issues/62815, this should probably just access the dashboard url generator directly from the dashboard plugin, rather than use the generic accessor.

Reply:
So export url generator on a dashboard plugin and use it in dashboard_enchanched? Correct? 
I'll go for it in follow up pr

Reply:
yep, and sg

Reply:
@stacey-gammon, pls review: https://github.com/elastic/kibana/pull/64628/